### ISSUE TITLE
Paraneue Gallery theme functions

### ISSUE THREAD

**Issue by aaronschachter** [DoSomethingArchive/legacy-website#3283]
- [x] Create function to return a Neue Gallery `<ul>`: http://neue.dosomething.org/#gallery.   Pseudo code:

```
/**
 * Returns <ul> of $items as gallery $type.
 */
function paraneue_get_gallery($items, $type = 'duo') {
  // Adds relevant classes to the <li> based on $type.
}
```
- [x] Create function to return a Neue Gallery tile: http://neue.dosomething.org/#tile.  Pseudo code:

```
/**
 * - title
 * - description
 * - image
 * - url
 */
function paraneue_get_tile($vars, $type = 'figure') {
  if ($vars['type'] == 'campaign') {
    return theme('paraneue_tile_campaign, $vars);
  }
  return theme('paraneue_tile_figure, $vars);
}
```
- [x] Use functions within the User Profile.  We'll want to add a `paranaue_dosomething_preprocess_user_profile` hook to set the function outputs as variables, and then print variables in the template.

**Comment by sbsmith86** [comment]
@aaronschachter Hey Aaron - What do you mean by "Use functions within User Profile"? This hook needs to live in template.php or is there a User Profile module it should be in?

**Comment by aaronschachter** [comment]
Hi @sbsmith86 !  In this `dosomething_user_preprocess_user_profile` function --  https://github.com/DoSomething/dosomething/blob/v0.3.78/lib/modules/dosomething/dosomething_user/dosomething_user.theme.inc#L62, the user's reportbacks are collected as the `$reportbacks` variable in the theme template: https://github.com/DoSomething/dosomething/blob/v0.3.78/lib/themes/dosomething/paraneue_dosomething/templates/user/user-profile.tpl.php#L52

There's also the `$campaigns` variable in there, but there's a little more work to do there, to render the Campaign data as a different tile.  I'm using the `campaign_block` theme function there as a placeholder. Actually gonna create an issue for that now as a todo.

**Comment by sbsmith86** [comment]
@aaronschachter Ah ok. So this new `paranaue_dosomething_preprocess_user_profile` hook can live in template.php and use the `$reportbacks` variable to generate the tiles using the new theme functions?

**Comment by aaronschachter** [comment]
yup!

**Comment by sbsmith86** [comment]
@aaronschachter For the `paraneue_get_gallery($items, $type = 'duo')` function, were you envisioning a theme function & tpl for every gallery type, similar to the get_tile() function? Or something else?

So we would have a .tpl for every gallery type and this function would choose the tpl to use based on type and then we pass the template an array of themed items.

**Comment by aaronschachter** [comment]
Yup, that's what I was thinking... but totally open to other ideas too.

**Comment by sbsmith86** [comment]
@aaronschachter my thought as well! just wanted to make sure I was on the same page.

**Comment by aaronschachter** [comment]
Closed by #3343
